src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -178,6 +178,7 @@ public void DiscoverTests(DiscoveryRequestPayload discoveryPayload, ITestDiscove
 
                     // Start the discovery of tests and wait for completion
                     this.currentDiscoveryRequest.DiscoverAsync();
+                    discoveryEventsRegistrar?.LogWarning(this.currentDiscoveryRequest.WarningMessage);
```

Comment:
Above is an async call, this will get called immediately. Shouldn't this be moved below the wait for completion ?